h2o-algos/src/main/java/hex/tree/SharedTreeModelWithContributions.java

```diff
@@ -31,25 +31,14 @@ public Frame scoreContributions(Frame frame, Key<Frame> destination_key) {
     return scoreContributions(frame, destination_key, null);
   }
 
-  private Frame removeSpecialColumns(Frame frame) {
-    Frame adaptFrm = new Frame(frame);
-    adaptTestForTrain(adaptFrm, true, false);
-    // remove non-feature columns
-    adaptFrm.remove(_parms._response_column);
-    adaptFrm.remove(_parms._fold_column);
-    adaptFrm.remove(_parms._weights_column);
-    adaptFrm.remove(_parms._offset_column);
-    return adaptFrm;
-  }
-
   @Override
   public Frame scoreContributions(Frame frame, Key<Frame> destination_key, Job<Frame> j) {
     if (_output.nclasses() > 2) {
       throw new UnsupportedOperationException(
               "Calculating contributions is currently not supported for multinomial models.");
     }
 
-    Frame adaptFrm = removeSpecialColumns(frame);
+    Frame adaptFrm = makeCompatibleWithTrainingFrame(frame);
```

Comment:
I personally find this name very confusing for 2 reasons:

- there is a method Frame#makeCompatible that has different semantics (it is about frame layout)
- training frame for supervised will have a response and possibly other special columns - I would not expect those to be deleted.

Reply:
How about "adaptTestWithoutSpecialCols"